### PR TITLE
CAPX-58: Added validation to importer.

### PR DESCRIPTION
**Pull request by RoSk0**
CAPX-58: Added validation to importer.

How to test:
- Create new content type
- Create new mapper with this content type
- Create 2+ importers with this mapper, do not import
- Remove content type created in the first step
- Click "Update profiles now" on importers page

### CODE REVIEW

File: includes/CAPx/Drupal/Mapper/EntityMapper.php [hunk 1]
```diff
@@ -196,5 +198,95 @@ public function mapFieldCollections($data) {
...
+    if ($entity_info) {
+      if (!isset($entity_info['bundles']) || empty($entity_info['bundles'][$bundle])) {
+        $mapper = $this->getMapper();
+        $message_key = $this->getImporter();
+        $message_vars = array(
+          '%mapper' => $mapper->label(),
+          '%entity_type' => $entity_info['label'],
+          '%bundle' => $bundle,
+        );
+        $message_text = t('Mapper %mapper was removed while been validated. Bundle %bundle, which it was configured to used, was removed from %entity_type entity.');
```
Comment (sherakama) [inline includes/CAPx/Drupal/Mapper/EntityMapper.php]: Please change to:

$message_text = t('Invalid bundle setting on %mapper. The bundle %bundle is no longer available and should either be restored, or the mapper %mapper should be deleted.');

File: includes/CAPx/Drupal/Mapper/EntityMapper.php [hunk 2]
```diff
@@ -196,5 +198,95 @@ public function mapFieldCollections($data) {
...
+    }
+    else {
+      $mapper = $this->getMapper();
+      $message_key = $this->getImporter();
+      $message_vars = array(
+        '%mapper' => $mapper->label(),
+        '%entity_type' => $entity_type,
+        '%bundle' => $bundle,
+      );
+      $message_text = t('Mapper %mapper was removed while been validated. Bundle %entity_type, which it was configured to use, was removed from the system.');
```
Comment (sherakama) [inline includes/CAPx/Drupal/Mapper/EntityMapper.php]: Please change to:

$message_text = t('Invalid entity setting on %mapper. The entity %entity_type is no longer available. Please restore the entity type or remove the mapper');

File: includes/CAPx/Drupal/Mapper/EntityMapper.php [hunk 3]
```diff
@@ -196,5 +198,95 @@ public function mapFieldCollections($data) {
...
+      capx_cfe_delete($mapper);
+
+      $importers = $this->getAffectedImporters();
+      $importer_links = array();
+      foreach ($importers as $importer) {
+        $importer_links[$importer->getMachineName()] = l(check_plain($importer->label()), 'admin/config/capx/importer/edit/' . $importer->getMachineName());
+      }
+      $message_vars['!importers'] = implode(', ', $importer_links);
+      $message_text .= ' ';
+      $message_text .= t('Importers that is using this mapper: !importers. You should delete them or select other mapper to use.');
```
Comment (sherakama) [inline includes/CAPx/Drupal/Mapper/EntityMapper.php]: Please change to:

$message_text .= t("The following importers are using an invalid mapper. Please update or delete the mapper settings: !importers);

File: includes/CAPx/Drupal/Mapper/EntityMapper.php [hunk 4]
```diff
@@ -196,5 +198,95 @@ public function mapFieldCollections($data) {
...
+        'message_vars' => $message_vars,
+        'importers' => $importer_links,
+      );
+      variable_set('stanford_capx_admin_messages', $messages);
+    }
+
+    return $entity_status;
+  }
+
+  public function getAffectedImporters() {
```
Comment (sherakama) [inline includes/CAPx/Drupal/Mapper/EntityMapper.php:375]: Please add function comments.

File: stanford_capx.pages.inc [hunk 5]
```diff
@@ -213,22 +213,30 @@ function stanford_capx_admin_config_import_edit($importer_machine_name) {
...
 
   // Nothing to see here.
   if (!$importer) {
-    throw new Exception(t("Could not find an importer by that name."));
+    $vars = array(
+      '%name' => $machine_name,
+      '!log' => l(t('log messages'), 'admin/reports/dblog'),
+    );
+    drupal_set_message(t('There was an issue loading the importer with %name machine name. Check !log.', $vars), 'error');
+    drupal_goto('admin/config/capx/importer');
```
Comment (sherakama) [inline stanford_capx.pages.inc]: Don't need this here.

File: stanford_capx.pages.inc [hunk 6]
```diff
@@ -213,22 +213,30 @@ function stanford_capx_admin_config_import_edit($importer_machine_name) {
...
   }
 
-  // Set the batch items.
-  $importer->createBatch();
+  if (!$importer->valid()) {
+    drupal_set_message(t('There was an issue running the importer.'), 'error');
+    drupal_goto();
```
Comment (sherakama) [inline stanford_capx.pages.inc]: Remove this.

File: stanford_capx.pages.inc [hunk 7]
```diff
@@ -213,22 +213,30 @@ function stanford_capx_admin_config_import_edit($importer_machine_name) {
...
+      '!log' => l(t('log messages'), 'admin/reports/dblog'),
+    );
+    drupal_set_message(t('There was an issue loading the importer with %name machine name. Check !log.', $vars), 'error');
+    drupal_goto('admin/config/capx/importer');
   }
 
-  // Set the batch items.
-  $importer->createBatch();
+  if (!$importer->valid()) {
+    drupal_set_message(t('There was an issue running the importer.'), 'error');
```
Comment (sherakama) [inline stanford_capx.pages.inc]: Please change message to: "There was an issue running the importer. Settings did not validate. Please check the error logs for more information."

File: includes/CAPx/Drupal/Mapper/EntityMapper.php (hunk 1 above)
Comment (RoSk0) [inline includes/CAPx/Drupal/Mapper/EntityMapper.php]: Fixed.

File: includes/CAPx/Drupal/Mapper/EntityMapper.php (hunk 2 above)
Comment (RoSk0) [inline includes/CAPx/Drupal/Mapper/EntityMapper.php]: Fixed.

File: includes/CAPx/Drupal/Mapper/EntityMapper.php (hunk 3 above)
Comment (RoSk0) [inline includes/CAPx/Drupal/Mapper/EntityMapper.php]: Fixed.

File: includes/CAPx/Drupal/Mapper/EntityMapper.php (hunk 4 above)
Comment (RoSk0) [inline includes/CAPx/Drupal/Mapper/EntityMapper.php:375]: Fixed.

File: stanford_capx.pages.inc (hunk 5 above)
Comment (RoSk0) [inline stanford_capx.pages.inc]: Fixed.

File: stanford_capx.pages.inc (hunk 6 above)
Comment (RoSk0) [inline stanford_capx.pages.inc]: Fixed.

File: stanford_capx.pages.inc (hunk 7 above)
Comment (RoSk0) [inline stanford_capx.pages.inc]: Fixed.

**Pull request merged**